src/SimpleSAML/Metadata/Sources/MDQ.php

```diff
@@ -217,8 +218,8 @@ private function writeToCache(string $set, string $entityId, array $data): void
 
         Logger::debug(sprintf('%s: Writing cache [%s] => [%s]', __CLASS__, $entityId, $cacheFileName));
 
-        /** @psalm-suppress TooManyArguments */
-        $this->fileSystem->appendToFile($cacheFileName, json_encode($data), true);
+        // using dumpFile instead of appendToFile here to replace any existing cache data
```

Comment:
Not sure this comment means anything when the previous code is gone. Probably it belongs in the commit message.